DEVGUIDE.md

```diff
@@ -188,6 +191,164 @@ To fix this, delete these folders:
 
 Where `<version>` corresponds to the latest Visual Studio version on your machine.
 
+## Performance and debugging
+
+Use the `Debug` configuration to test your changes locally. It is the default. Do not use the `Release` configuration! Local development and testing of Visual Studio tooling is not designed for the `Release` configuration.
+
+### Writing and running benchmarks
+
+Existing compiler benchmarks can be found in `tests\benchmarks\`.
+
+### Benchmarking and profiling the compiler
+
+**NOTE:** When running benchmarks or profiling compiler, and comparing results with upstream version, make sure:
+
+* Always build both versions of compiler/FCS from source and not use pre-built binaries from SDK (SDK binaries are crossgen'd, which can affect performance).
+* To run `Release` build of compiler/FCS.
+
+### Example benchmark setup using [BenchmarkDotNet](https://github.com/dotnet/BenchmarkDotNet)
+
+1. Perform a clean build of the compiler and FCS from source(as described in this document).
```

Comment:
Would that be `.\Build.cmd -noVisualStudio`, or do you need some other bits as well?

Reply:
Added a note there.